scalopus_tracing/src/native/tracepoint_collector_native.h

```diff
@@ -42,15 +44,17 @@ namespace tracepoint_collector_types
 //! Timepoint of that clock.
 using TimePoint = uint64_t;
 //! Trace event as it is stored in the ringbuffer.
-using StaticTraceEvent = std::tuple<TimePoint, unsigned int, uint8_t>;
+using StaticTraceEvent = std::tuple<TimePoint, unsigned int, uint8_t, nlohmann::json>;
```

Comment:
We should use a `std::unique_ptr<scalopus::Data>` as dynamic data payload. That allows preventing the necessity to require the `nlohmann::json` dependency in the process under test. Using a `unique_ptr` is the lightest thing I can think of. It default initialises to a `nullptr`, which is no dynamic data associated to the event. Or it can point to a payload which we can directly convert into cbor for the consumer side to interpret in whatever way it wants.

Not caused by this PR, but reading this back now it would've been much better if I had used some `using` things for the `unsigned int` and `uint8_t` here.